extension/src/popup/views/RecoverAccount/index.tsx

```diff
@@ -26,41 +31,90 @@ import { PasswordRequirements } from "popup/components/PasswordRequirements";
 
 import "./styles.scss";
 
-export const RecoverAccount = () => {
-  const publicKey = useSelector(publicKeySelector);
-  const authError = useSelector(authErrorSelector);
-  const publicKeyRef = useRef(publicKey);
+interface PhraseInputProps {
+  phraseInput: string;
+  index: number;
+  isDisabled: boolean;
+  handleMnemonicInputChange: (value: string, index: number) => void;
+}
+
+const PhraseInput = ({
+  phraseInput,
+  index,
+  isDisabled,
+  handleMnemonicInputChange,
+}: PhraseInputProps) => {
+  const [isTextShowing, setIsTextShowing] = useState(false);
+  const [inputValue, setInputValue] = useState("");
 
+  return (
+    <div
+      key={phraseInput}
+      className={`RecoverAccount__phrase-input ${
+        isDisabled ? "RecoverAccount__phrase-input--disabled" : ""
+      }`}
+    >
+      <div className="RecoverAccount__phrase-input__number">{index + 1}. </div>
+      <Input
+        autoComplete="off"
+        disabled={isDisabled}
+        id={phraseInput}
+        name={phraseInput}
+        onChange={(e) => {
+          handleMnemonicInputChange(e.target.value, index);
+          setInputValue(e.target.value);
+        }}
+        onPaste={(e) => e.preventDefault()}
+        placeholder="Enter your 12 word phrase to restore your wallet"
+        type={isTextShowing ? "text" : "password"}
+        value={inputValue}
+      />
+      <div
+        className="RecoverAccount__password-toggle"
+        onClick={() => setIsTextShowing(!isTextShowing)}
+      >
+        {isTextShowing ? <Icon.Eye /> : <Icon.EyeOff />}
+      </div>
+    </div>
+  );
+};
+
+export const RecoverAccount = () => {
   interface FormValues {
     password: string;
     confirmPassword: string;
-    mnemonicPhrase: string;
     termsOfUse: boolean;
   }
 
   const initialValues: FormValues = {
     password: "",
     confirmPassword: "",
-    mnemonicPhrase: "",
     termsOfUse: false,
   };
 
+  const publicKey = useSelector(publicKeySelector);
+  const authError = useSelector(authErrorSelector);
+  const publicKeyRef = useRef(publicKey);
   const RecoverAccountSchema = YupObject().shape({
-    mnemonicPhrase: mnemonicPhraseValidator,
     password: passwordValidator,
     confirmPassword: confirmPasswordValidator,
     termsOfUse: termsofUseValidator,
   });
 
   const dispatch = useDispatch();
+  const phraseLengthOptions = [12, 24];
```

Comment:
I imagine this list will be longer - I've only tested with 12 and 24 length phrases so far